Changelog.txt

```diff
@@ -3442,13 +3442,130 @@ Additionally, the problem of zig-zag issue following in the South direction has
+	SPELLFLAG_TARG_ONLYSELF should be added to defs.
+	//magery_9 "You can only use this spell on yourself." could be added to defmessages.
+Changed: SERV.LOG command changed to support console colors. (Issue: [Feature Request] Colored Serv & Src Log System #1158)
+	SERV.LOG works same as before.
+	Usage of custom color system is: SERV.LOG @color,type,mask message
+	Example: SERV.LOG @CTCOL_RED,LOGL_EVENT,LOGM_NOCONTEXT|LOGM_KILLS message
+	LOGM, LOGL and CTCOL values should be added to defs.
+
+25-03-2024, xwerswoodx
+Added: Sphere.ini setting CanPetsDrinkPotion (true in default) to make pets drink potion when their owner drop it on it. (Issue: #1147)
+	CanPetsDrinkPotion = 0 can be added to sphere.ini to disable that.
+Fixed: Missing fCheckOnly checks block the taming skill. (Issue: #1219)
+
+26-03-2024, xwerswoodx
+Added: New trigger @Drink added for characters. (Issue: #1162)
+	I: The character drinking the item.
+	SRC: The character drinking the item.
+	ARGO = the item that drank
+	ARGN1 = cooldown (in seconds)
+	ARGN2 = consumption amount (1 in default)
+	LOCAL.BottleID = empty bottle id that will be given when after item drank. (Will be given ARGN2 amount of empty bottle in default)
+	- Returns:
+	1: Stops character from drinking the item.
+	5: Let character to drink but will only give 1 empty bottle.
+	6: Let character to drink but will not give any empty bottle.
+	- Messages:
+	DRINK_NOT_ENOUGH, "You don't have enough %s to drink this."
+Added: New trigger @AfkMode added for characters. (Issue: #1161)
+	I: The character changing AFK status.
+	SRC: The character changing AFK status.
+	ARGN1 = Current Afk Status
+	ARGN2 = Next Afk Status
+	- Returns:
+	1: Stops character from switching afk status.
+Added: New trigger @Reveal added for characters. (Issue: #1157)
+	I: The character who is revealing.
+	SRC: The character who is revealing.
+	- Returns:
+	1: Prevents character from being revealed.
+Added: New triggers @ArrowQuest_Add and @ArrowQuest_Close added for characters. (Issue: #1116)
+	I: The character who activating/deactivating arrow quest.
+	SRC: The character who activating/deactivating arrow quest.
+Added: New function triggers f_onaccount_block and f_onaccount_unblock. (Issue: #1156)
+	ARGS = Account name.
+	- Returns:
+	1: Prevents account from blocking/unblocking.
+Added: New function trigger f_onserver_broadcast. (Issue: #1145)
+	ARGS = message
+	- Returns:
+	1: Prevents server to send broadcast message.
+Added: Two new LOCAL variables ItemPoisonReductionChance and ItemPoisonReductionAmount to @Hit trigger. (Issue: #1159)
+	LOCAL.ItemPoisonReductionChance = Is the chance that weapon reduce poison morez. (Default 100%)
+	LOCAL.ItemPoisonReductionAmount = Is the amount that weapon reduce. (Default: CurrentPoisonAmount/ 2)
+Added: You can now use TDATA2 for setting default cooldown (delay) for drinks. (Default: 15secs for potions, 1500 secs for ales)
```

Comment:
We can add default cooldown to ini for drinks, but it's different for ales and potions, but I can put some extra integers in sphere.ini for default cooldowns.

Reply:
I feel it would be wise to have a default cooldown for each type (maybe in the typedef? or ini?), then give the possibility to overwrite for the specific item. In the latter case, imho it would be better to store the info in a tag, so that it can be statically added to the typedef but also changed at runtime for maximum flexibility.

Reply:
Also, could you please explain differently in the changelog what do the new locals do? I'm not sure i've grasped it fully

Reply:
> I feel it would be wise to have a default cooldown for each type (maybe in the typedef? or ini?), then give the possibility to overwrite for the specific item. In the latter case, imho it would be better to store the info in a tag, so that it can be statically added to the typedef but also changed at runtime for maximum flexibility.

Honestly I added TDATA2 exactly for this, it works well with @Drink trigger, I mean, TDATA2 is something that you can set under [ITEMDEF] for any drink to define it's cooldown timer, but it can be override under @Drink trigger dynamically, but if you still wish to add another tag that can be defined under [itemdef] and overridable dynamically, we can add TAG.OVERRIDE, https://wiki.spherecommunity.net/index.php?title=Override_TAGs like in here.

For LOCALS, they were for the POISONED WEAPONS, poison on weapon saved on morez, and it drops some value from it on every hit of player, normally this is 100%. but I added CHANCE and AMOUNT override to let gms decide how this poison effect removes, seems like I haven't put much info in changelog, I can improve it.